assets/css/abstracts/_mixins.scss

```diff
@@ -1,10 +1,14 @@
-// Converts a font-size in px to rem and optionally sets a relative line-height.
-@mixin font-size($sizeValue: 16px, $lineHeight: false) {
-	font-size: ($sizeValue / 16px) * 1rem;
-
-	@if ($lineHeight) {
-		line-height: $lineHeight / $sizeValue;
-	}
+$fontSizes: (
+	"smaller": 0.75,
+	"small": 0.875,
+	"regular": 1,
+	"large": 1.25,
+	"larger": 2,
```

Comment:
I think `smaller` `small`... would fit well with sizing but not much with font sizes, as they provide no obvious context, I would possibly suggest something like `heading` `caption` `body` and so on, they provide more context.

see examples in:
- [Polaris](https://github.com/Shopify/polaris-react/blob/master/src/styles/foundation/_typography.scss#L17-L56)
- [Material Design](https://github.com/material-components/material-components-web/blob/master/packages/mdc-typography/_variables.scss#L166-L178)

However, there’s no size fit all, other opinions on this:
- [Font size naming convention in designer news](https://www.designernews.co/stories/105780-font-size-naming-convention).
- [Naming typography patterns in CSS](https://keenanpayne.com/blog/naming-typography-patterns-css.html) 
- [Typography in Design Systems](https://superfriendlydesign.systems/articles/typography-in-design-systems/)
- [CSS utility classes for sizing and naming conventions](https://catalin.red/css-utility-classes-naming-conventions/)

### Why do I care so much about such a small thing?

Because those names will in the future be a public API for our blocks, both in global styles, extending and our style guide, getting them right is somehow a bit important.

This is by no means a blocker, we can address this later, we can merge this, or we can address it now, your call.

Reply:
Loads of good ideas here, thanks for the feedback @senadir. So if I understand you correctly, you are proposing to define a `heading` named font size, for example, so if we ever get a `--wp-gs-heading-font-size` custom property (or whichever system global styles ends up using), we could easily update the font sizes of all headings to use the style from global styles. Is that correct?

While I really like this idea, I have a couple of thoughts about this:
* Currently, if we have a heading or a button that should inherit the theme font size, I think it shouldn't have any font size defined in its CSS and instead rely on the inherited value. Said in other words: all the font sizes introduced in this PR are variations of the body text font size. If an element must honor the theme styles (for example, an `<h2>` needs to render as all other `<h2>` are rendered in that theme), it should just inherit the font size. At least, until global styles land.
* In parallel to this, I think it's a bit too early to know how the final implementation of global styles will be and which specific values we will be able to use. In the examples I have seen, we would be able to read the heading, body and button font sizes, for example. But what about caption, for example? There are dozens of HTML elements and I don't think global styles will expose specific styles for all of them, so my position with this would be to wait a bit more to see how global styles evolve before embracing this.

Let me know how that sounds! In any case, I agree we shouldn't block this PR because of that since it can be tackled in a follow-up.

Reply:
Well, I'm not suggesting 1:1 with GS, global styles could be an example but it’s still early for it, my point is that naming font sizes large, small... provides no context to us as developers or to third-party developers, font size depends on context, you will probably never have a heading that is 10px, having semantic names will make sure we're using correct tags and correct placement.
Global styles could be another layer of abstraction above this if we don't want to expose everything, I don't disagree on the values or how they're used, mainly the naming and how intuitive it can get.

Reply:
> I think smaller small... would fit well with sizing but not much with font sizes, as they provide no obvious context, I would possibly suggest something like heading caption body and so on, they provide more context.

Isn't the point here to have a flexible system where the content is unknown? Is we based it on selectors we could run into code smell where the design uses one size, and we're forced to use a named size that doesn't match our usage.

Reply:
> Isn't the point here to have a flexible system where the content is unknown

A design system still needs to follow certain constraints 

> If we based it on selectors we could run into code smell where the design uses one size, and we're forced to use a named size that doesn't match our usage.

We're not basing it on selector but on context, a body-sized text could be any element, not strictly applied to the body tag, by having those rules, a designer or a developer will ask the question "Is this a heading? if not, why is the size the same as other headings within this design", if we're going to communicate and emphasize importance and context via fonts sizes, variables should be named accordingly.

Reply:
@senadir Not all headers are equal. There are also times where a heading may be styled differently right? I think the defined sizes work better for our use case. We have a base and everything else is just a slight variation of that. Perhaps we just need inline docs to identify which are most commonly used for headings?

Reply:
If by headings we are referring to `<h1>`, `<h2>`, etc., I don't think we need to establish a relationship between them and specific font sizes, in several occasions we need to display headings as if they were regular text.

In the other hand, for headings that need to honor the font size defined by the theme (I'm thinking on the Filter block titles, for example) I think not setting any font size on our end and letting styles cascade is the best option.
![imatge](https://user-images.githubusercontent.com/3616980/83039741-cc0c5c00-a03e-11ea-8038-98ea679c02ff.png)